src/main/java/treechopper/common/tree/Tree.java

```diff
@@ -2,49 +2,54 @@
 
 import net.minecraft.util.math.BlockPos;
 
-import java.util.HashSet;
-import java.util.Set;
-
+import java.util.ArrayList;
+
+/**
+ * POJO Tree.
+ *
+ * TODO:
+ *  Add sapling plant location function
```

Comment:
I'm thinking of adding some logic around placing the saplings, such as finding a block of dirt somewhere surrounding the tree. Currently it will try to place a sapling at origin + 1,0,1 regardless of if there is a valid dirt block for the sapling to sit on.  I think the logic needs to search the surrounding blocks in a 3x3x3 square, sort of like how `analyzeTree` works.

Reply:
I see. Yeah exactly, sapling will be placed on 1 0 1 from origin. If there is dirt under, sapling will be placed correctly. Otherwise sapling will be just laying on top of block, like you would throw it on a ground on a position 1 0 1. To check area around origin and place it on first founded dirt block is nice to have feature :) My plans were to set position of placing sapling in a config, but maybe in a future.